draft-ietf-mls-protocol.md

```diff
@@ -857,19 +843,22 @@ Section 6 of {{RFC7748}}.  If implementers use an alternative
 implementation of these elliptic curves, they SHOULD perform the
 additional checks specified in Section 7 of {{RFC7748}}
 
-### P-256, SHA-256, and AES-128-GCM
+#### X448
 
-This ciphersuite uses the following primitives:
-
-* Hash function: SHA-256
-* AEAD: AES-128-GCM
+Given an octet string X, the private key produced by the
+Derive-Key-Pair operation is SHA-512(X) truncated to 448 bits (Recall that any 56-octet
```

Comment:
Should we use HKDF instead of SHA directly?  In other words, make node secrets variable-length, where the length is determined by the KEM in use.

* P256, X25519: 32 bytes
* X448: 56 bytes
* P521: 66 bytes

As a bonus, that would save a couple of hash invocations.

Reply:
Unless there is a specific reasons not to do it (cc @beurdouche), I think it would be great to harmonise this. Having different hash algorithms in here makes no sense to me if it can be avoided.